src/code/z_eff_blure.c

```diff
@@ -66,40 +68,977 @@ void func_800A81F0(EffectBlure* this, Vec3f* p1, Vec3f* p2) {
+        return 0;
+    }
+
+    while (true) {
+        if (this->elements[0].state == 0) {
+            for (i = 0; i < 15; i++) {
+                this->elements[i] = this->elements[i + 1];
+            }
+
+            this->elements[i].state = 2;
+            this->elements[i].p1.x = 0;
+            this->elements[i].p1.y = 0;
+            this->elements[i].p1.z = 0;
+            this->elements[i].p2.x = 0;
+            this->elements[i].p2.y = 0;
+            this->elements[i].p2.z = 0;
+            this->elements[i].flags = 0;
+            this->elements[i].timer = 0;
+
+            this->numElements--;
+            if (this->numElements <= 0) {
+                this->numElements = 0;
+                return 0;
+            }
+        } else {
+            break;
+        }
+    }
+
+    if (this->elements[0].state == 2) {
+        return 0;
+    }
+
+    for (i = 0; i < this->numElements; i++) {
+        this->elements[i].timer++;
+    }
+
+    if (this->elemDuration < this->elements[0].timer) {
+        for (i = 0; i < 15; i++) {
+            this->elements[i] = this->elements[i + 1];
+            if (this->elements[i].timer >= this->elemDuration) {
+                this->elements[i].timer = this->elemDuration;
+            }
+        }
+
+        this->elements[i].state = 2;
+        this->elements[i].p1.x = 0;
+        this->elements[i].p1.y = 0;
+        this->elements[i].p1.z = 0;
+        this->elements[i].p2.x = 0;
+        this->elements[i].p2.y = 0;
+        this->elements[i].p2.z = 0;
+        this->elements[i].flags = 0;
+        this->elements[i].timer = 0;
+
+        this->numElements--;
+        if (this->numElements <= 0) {
+            this->numElements = 0;
+            return 0;
+        }
+        return 0;
+    }
+
+    for (i = 0; i < this->numElements; i++) {
+        if (this->elements[i].timer >= this->elemDuration) {
+            this->elements[i].timer = this->elemDuration;
+        }
+    }
+
+    this->addAngle += this->addAngleChange;
+    return 0;
+}
+
+void EffectBlure_UpdateFlags(EffectBlureElement* elem) {
+    Vec3f sp64;
+    Vec3f sp58;
+    Vec3f sp4C;
+    Vec3f sp40;
+    EffectBlureElement* prev = elem - 1;
+    EffectBlureElement* next = elem + 1;
+    f32 sp34;
+    f32 sp30;
+    f32 sp2C;
+
+    if (((elem - 1)->state == 0) || ((elem + 1)->state == 0)) {
+        elem->flags &= ~3;
+        elem->flags |= 2;
+    } else {
+        Math_Vec3s_DiffToVec3f(&sp64, &elem->p1, &prev->p1);
+        Math_Vec3s_DiffToVec3f(&sp58, &elem->p2, &prev->p2);
+        Math_Vec3s_DiffToVec3f(&sp4C, &next->p1, &elem->p1);
+        Math_Vec3s_DiffToVec3f(&sp40, &next->p2, &elem->p2);
+
+        if (Math3D_AngleBetweenVectors(&sp64, &sp4C, &sp34) || Math3D_AngleBetweenVectors(&sp58, &sp40, &sp30) ||
+            Math3D_AngleBetweenVectors(&sp4C, &sp40, &sp2C)) {
+            elem->flags &= ~3;
+            elem->flags |= 0;
+        } else if ((sp34 <= -0.5f) || (sp30 <= -0.5f) || (sp2C <= 0.7071f)) {
```

Comment:
```suggestion
        } else if ((sp34 <= -0.5f) || (sp30 <= -0.5f) || (sp2C <= 0.7071f)) { // sp2C <= sqrt(2) / 2
```
I wonder if this comment is necessary or if there's a cleaner way to represent this

Reply:
I feel like it would a bit weird to have a comment specifically for this part of the `||`, but if there is a cleaner way to write the float, then we should definitely use it. I don't have any idea myself tho

Reply:
Yeah that's fine, I was just throwing an idea out there. I can't think of any good idea myself